Dockerfile

```diff
@@ -122,7 +122,18 @@ RUN --mount=target=. --mount=target=/root/.cache,type=cache \
   set -ex
   xx-go build ${GOBUILDFLAGS} -gcflags="${GOGCFLAGS}" -ldflags "$(cat /tmp/.ldflags) -extldflags '-static'" -tags "osusergo netgo static_build seccomp ${BUILDKITD_TAGS}" -o /usr/bin/buildkitd ./cmd/buildkitd
   xx-verify ${VERIFYFLAGS} /usr/bin/buildkitd
-  if [ "$(xx-info os)" = "linux" ]; then /usr/bin/buildkitd --version; fi
+  if [ "$(xx-info os)" = "linux" ]; then
+    set +e
+    timeout 3 /usr/bin/buildkitd --version
+    exitcode=$?
+    if [ $exitcode -eq 124 ]; then
```

Comment:
some comments to understand these exit codes could help. I think we shouldn't just ignore timeout,  but maybe if `xx-info is-cross` then warn and retry again for couple of times.